src/QuadForm/Quad/Spaces.jl

```diff
@@ -613,13 +612,13 @@ end
 function _quadratic_form_with_invariants(dim::Int, det::nf_elem, finite::Vector, negative::Dict{<:InfPlc, Int})
   @hassert :Lattice 1 dim >= 1
   @hassert :Lattice 1 !iszero(det)
-  K::AnticNumberField = parent(det)
+  K = parent(det)
```

Comment:
This was added for type stability. Have you checked that everything is still fine?

Reply:
I will have to re-put it. There are very odd things happening in this function, which I barely understand on the `code_warntype`. The variable `finite0` is defined and used only at the very end of the function in a `hassert`. If we remove the `hassert` and infer the type for `finite0`, for instance
```
local finite0::Vector{<: NfOrdIdl}
finite0 = copy(finite)
```
or
```
local finite0::typeof(finite)
finite0 = copy(finite)
```
or 
```
finite0 = eltype(finite)[p for p in finite]
```
in all cases, if we remove the only line calling this variable, then `code_warntype` will still show `finite0::Any` in red...

Reply:
This is all related to https://github.com/JuliaLang/julia/issues/15276. Every time one uses a `p -> begin; x, y, z; end`, every "captured" variable which is used (like `x`, `y`, `z`), might become unstable in the surrounding function. This also effects generators like `(p + z for p in 1:10)`. The `let` is a workaround, also mentioned in the issue above.

Reply:
I see, I completely forgot about this. So maybe the `code_warntype` might not capture the issue. Since it was a recent fix I will put this back too until we make sure the problem is solved (to avoid any performance issue)